include/cantera/kinetics/Arrhenius.h

```diff
@@ -0,0 +1,339 @@
+                                   const UnitSystem& units,
+                                   const UnitStack& rate_units);
+    using ReactionRate::setParameters;
+
+    virtual void getParameters(AnyMap& node) const;
+
+    //! Return parameters - required for legacy framework
+    //! @todo: merge with single-parameter version after removal of old framework
+    virtual void getParameters(AnyMap& node, const Units& rate_units) const;
+
+    void check(const std::string& equation, const AnyMap& node);
+
+    //! Return the pre-exponential factor *A* (in m, kmol, s to powers depending
+    //! on the reaction order)
+    double preExponentialFactor() const {
+        return m_A;
+    }
+
+    //! Return the temperature exponent *b*
+    double temperatureExponent() const {
+        return m_b;
+    }
+
+    //! Return the intrinsic activation energy *Ea* [J/kmol]
+    //! The value corresponds to the constant specified by input parameters;
+    //! in the simplest case, this is equal to the activation energy itself
+    double intrinsicActivationEnergy() const {
+        return m_Ea_R * GasConstant;
+    }
+
+    // Return units of the reaction rate expression
+    const Units& rateUnits() const {
+        return m_rate_units;
+    }
+
+    //! Return reaction order associated with the reaction rate
+    double order() const {
+        return m_order;
+    }
+
+    //! Set units of the reaction rate expression
+    void setRateUnits(const UnitStack& rate_units) {
+        if (rate_units.size() > 1) {
+            m_rate_units = rate_units.product();
+            m_order = 1 - m_rate_units.dimension("quantity");
+        } else {
+            m_order = NAN;
+            m_rate_units = rate_units.standardUnits();
+        }
+    }
+
+    //! Get buffered value of third-body concentration
+    const bool allowNegativePreExponentialFactor() const {
+        return m_negativeA_ok;
+    }
+
+    //! Get buffered value of third-body concentration
+    void setAllowNegativePreExponentialFactor(bool value) {
+        m_negativeA_ok = value;
+    }
+
+
+protected:
+    bool m_negativeA_ok; //!< Flag indicating whether negative A values are permitted
+    double m_A; //!< Pre-exponential factor
+    double m_b; //!< Temperature exponent
+    double m_Ea_R; //!< Activation energy (in temperature units)
+    double m_order; //!< Reaction order
+    Units m_rate_units; //!< Reaction rate units
+};
+
+
+//! Arrhenius reaction rate type depends only on temperature
+/*!
+ * A reaction rate coefficient of the following form.
+ *
+ *   \f[
+ *        k_f =  A T^b \exp (-Ea/RT)
+ *   \f]
+ *
+ * @todo supersedes Arrhenius: replace existing instances with this class, rename,
+ *      and deprecate Arrhenius3.
+ *
+ * @ingroup arrheniusGroup
+ */
+class ArrheniusRate final : public ArrheniusBase
+{
+public:
+    ArrheniusRate() = default;
+    using ArrheniusBase::ArrheniusBase; // inherit constructors
+    using ArrheniusBase::setParameters;
```

Comment:
Likewise, this `using` is now unnecessary since `ArrheniusRate` overrides the only overload of `setParameters` as it is.